azurelinuxagent/ga/collect_logs.py

```diff
@@ -170,29 +175,33 @@ def _collect_logs(self):
         # Invoke the command line tool in the agent to collect logs, with resource limits on CPU and memory (RAM).
 
         systemd_cmd = [
-            "systemd-run", "--unit={0}".format(logcollector.CGROUPS_UNIT),
+            "systemd-run", "--property=CPUAccounting=no", "--property=MemoryAccounting=no",
+            "--unit={0}".format(logcollector.CGROUPS_UNIT),
             "--slice={0}".format(cgroupconfigurator.LOGCOLLECTOR_SLICE), "--scope"
         ]
 
         # The log tool is invoked from the current agent's egg with the command line option
         collect_logs_cmd = [sys.executable, "-u", sys.argv[0], "-collect-logs"]
         final_command = systemd_cmd + collect_logs_cmd
 
-        def exec_command(output_file):
+        def exec_command():
             start_time = datetime.datetime.utcnow()
             success = False
             msg = None
             try:
-                shellutil.run_command(final_command, log_error=False, stdout=output_file, stderr=output_file)
-                duration = elapsed_milliseconds(start_time)
-                archive_size = os.path.getsize(COMPRESSED_ARCHIVE_PATH)
-
-                msg = "Successfully collected logs. Archive size: {0} b, elapsed time: {1} ms.".format(archive_size,
-                                                                                                    duration)
-                logger.info(msg)
-                success = True
-
-                return True
+                base_dir = conf.get_lib_dir()
+                with tempfile.TemporaryFile(dir=base_dir, mode="w+b") as stdout:
```

Comment:
why do we need those tmp files?

Reply:
Nothing specific, we are doing same thing for extension systemd-run commands. Since it's working solution, I decided to use same thing.

Reply:
In the case of extensions commands we use files because we need to reset stdout/stderr when systemd-run fails

```
           # Reset the stdout and stderr
            stdout.truncate(0)
            stderr.truncate(0)

```
but here the files are not needed

Reply:
I see your point, I use default Subprocess.PIPE instead.